source/slang/slang-emit-spirv.cpp

```diff
@@ -2960,24 +2961,19 @@ struct SPIRVEmitContext
         SpvExecutionMode mode = SpvExecutionModeMax;
         for (auto globalInst : referencedBuiltinIRVars)
         {
-            if (auto thisMode = isDepthOutput(globalInst))
+            auto thisMode = getDepthOutputExecutionMode(globalInst);
+            if (mode == SpvExecutionModeMax)
+                mode = thisMode;
+            else if (mode != thisMode)
             {
                 needDepthReplacingMode = true;
-                if (mode == SpvExecutionModeMax)
-                    mode = thisMode;
-                else if (mode != thisMode)
-                    mode = SpvExecutionModeDepthReplacing;
+                mode = SpvExecutionModeDepthReplacing;
                 break;
             }
         }
         if (!needDepthReplacingMode)
             return;
-        emitOpExecutionMode(getSection(SpvLogicalSectionID::ExecutionModes),
-            nullptr,
-            entryPoint,
-            SpvExecutionModeDepthReplacing);
-        if (mode != SpvExecutionModeDepthReplacing &&
-            mode != SpvExecutionModeMax)
+        if (mode != SpvExecutionModeMax)
```

Comment:
```suggestion
        if (mode == SpvExecutionModeMax)
            return;
```
purely a syntax candy suggestion: removes need for the indentation & braces below